File: TrincaBarbecue.SharedKernel/Specification/AndSpecification.cs
namespace TrincaBarbecue.SharedKernel.Specification
{
    public class AndSpecification<T> : AbstractSpecification<T>
    {
        private ISpecification<T> _left;
        private ISpecification<T> _right;

        public AndSpecification(ISpecification<T> left, ISpecification<T> right) : base()
        {
            _left = left;
            _right = right;
        }

        public override bool IsSatisfied(T entity)
        {
            return _left.IsSatisfied(entity) && _right.IsSatisfied(entity);
        }
    }
}

File: SummitPro.Web/Scripts/Specification/Operators/AndSpecification.ts
import ISpecification from "../Contracts/ISpecification";

export default class AndSpecification<T> implements ISpecification<T> {
    private left: ISpecification<T>;
    private right: ISpecification<T>;

    constructor(left: ISpecification<T>, right: ISpecification<T>) {
        this.left = left;
        this.right = right;
    }

    isSatisfiedBy(item: T): boolean {
        return this.left.isSatisfiedBy(item) && this.right.isSatisfiedBy(item);
    }
}
